wgpu-core/src/device/resource.rs

```diff
@@ -1357,7 +1361,7 @@ impl Device {
             } else {
                 wgt::TextureUses::RESOURCE
             };
-            texture.hal_usage & mask_copy & mask_dimension & mask_mip_level
+            texture.hal_usage & resolved_hal_usage & mask_copy & mask_dimension & mask_mip_level
```

Comment:
nitpick: I suspect it's actually incorrect to keep using `texture.hal_usage` in this conjunctive clause, though the consequences of being incorrect here are pretty easy to debug (the platform yells that you aren't allowed to do the thing you never said you'd do). I'll try running tests without it, and if that works, I'll squash-and-merge.